scripts/partition_manager/partition_manager.rst

```diff
@@ -0,0 +1,408 @@
+   It cannot be used by phony partitions.
+   The list can contain any kind of partition.
+
+   Note that if the target partition is the ``app`` or a partition that spans over the ``app``, the size is effectively split between them, because the size of the ``app`` is dynamically decided.
+
+.. _pm_yaml_preprocessing:
+
+Configuration file preprocessing
+================================
+
+Each :file:`pm.yml` file is preprocessed to resolve symbols from Kconfig and DTS.
+
+The following example shows what kind of preprocessing is possible:
+
+.. code-block:: yaml
+
+   #include <autoconf.h>
+
+   # 'b0' is the name of the image partition.
+   b0:
+
+     # b0 is placed before the mcuboot partition if the mcuboot partition
+     # exists, otherwise it is stored before the app partition.
+     placement: {before: [mcuboot, app]}
+
+     # The size of the b0 partition is configured in Kconfig.
+     size: CONFIG_BOOTLOADER_PARTITION_SIZE
+
+   # Don't define the provision partition if the SoC is nRF9160, because
+   # the provisioning data is stored in the UICR->OTP data region.
+
+   #ifndef CONFIG_SOC_NRF9160
+
+   # 'provision' is the name of the placeholder partition.
+   provision:
+     # This partition is stored at the very end of flash.
+     placement: last
+
+   #endif /* CONFIG_SOC_NRF9160 */
+
+.. _pm_build_system:
+
+Build system
+************
+The Partition Manager uses Zephyr's multi-image build system to collect configurations for all sub-images.
+If one or more sub-images are included in a build, a set of properties for that sub-image is appended to a global list.
+
+These properties are:
+
+   * Path to :file:`pm.yml`
+   * Build directory path
+   * Path to generated include folder
+
+After CMake finishes configuring the sub-images, the Partition Manager script is executed in configure time (``execute_process``) with the aforementioned list as argument.
+The configurations generated by the Partition Manager script are imported as CMake variables (see :ref:`pm_cmake_usage`).
+
+.. _pm_generated_output_and_usage:
+
+Generated output
+================
+For each sub-image and the root application, Partition Manager generates three files: a C header file :file:`pm_config.h`, a Kconfig file :file:`pm.config`, and a YAML file :file:`pm_static.yml`.
+
+The C header file is used in the C code, while the Kconfig file is imported in CMake.
+Both these files contain the start address and size of all partitions.
+The Kconfig file additionally contains the build directory and generated include folder for each image.
+
+The YAML file contains the internal state of the Partition Manager at the end of processing.
+This means it contains the merged contents of all :file:`pm.yml` files, the sizes and addresses of all partitions, and other information generated by the Partition Manager.
+
+Usage
+=====
+When using Partition Manager in your application, you should be aware of the functionality it adds.
```

Comment:
That sentence doesn't say anything. What is it that we describe in the following sections?

Reply:
The following sections describe the output generated by the partition manager.

Reply:
Not really - that is the section before. This section shows what you can do with that output - correct?

Updated to "The output that Partition Manager generates can be used in various areas of your code." for now.

Reply:
That is fine by me. I mostly a sentence before having the sub-sections. This is something I learned to do in high-school, so it might not be applicable any more :)